docs/data-management/automatic-compaction.md

```diff
@@ -203,6 +203,82 @@ The following auto-compaction configuration compacts updates the `wikipedia` seg
 }
 ```
 
+## Concurrent compaction
+
+:::info
+Concurrent compaction is an [experimental feature](../development/experimental.md) and is not currently available for SQL-based ingestion.
+:::
+
+If you enable automatic compaction, you can also use concurrent compaction for streaming and legacy JSON-based batch ingestion. Concurrent compaction compacts the data as you ingest it.
+
+Setting up concurrent compaction is a two-step process. The first is to update your datasource and the second is to update your ingestion job.
+
+Using concurrent compaction in the following scenarios can be beneficial:
+
+- If the job with an `APPEND` task and the job with a `REPLACE` task have the same segment granularity. For example, when a datasource and its streaming ingestion job have the same granularity.
+- If the job with an `APPEND` task  has a finer segment granularity than the replacing job.
+
+We do not recommend using concurrent compaction when the job with an `APPEND` task has a coarser granularity than the job with a `REPLACE` task. For example, if the `APPEND` job has a yearly granularity and the `REPLACE` job has a monthly granularity. The job that finishes second will fail.
+ 
+### Configure concurrent compaction
+
+##### Update the compaction settings with the API
+ 
+ First, prepare your datasource for concurrent compaction by setting its task lock type to `REPLACE`.
+Add the `taskContext` like you would any other auto-compaction setting through the API:
+
+```shell
+curl --location --request POST 'http://localhost:8081/druid/coordinator/v1/config/compaction' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "dataSource": "YOUR_DATASOURCE",
+    "taskContext": {
+        "taskLockType": "REPLACE"
+    }
+}'
+```
+
+##### Update the compaction settings with the UI
+
+In the **Compaction config** for a datasource, set  **Allow concurrent compaction append tasks** to **True**.
```

Comment:
The UI for compaction says "Allow concurrent compactions (experimental)"
<img width="563" alt="image" src="https://github.com/apache/druid/assets/95239065/7e162ef7-35cc-4704-bbe8-9eb1a2a8e22f">